packages/cli/commands/project/watch.js

```diff
@@ -0,0 +1,268 @@
+
+    try {
+      await queueBuild(accountId, projectName, currentBuild.get());
+      logger.debug(i18n(`${i18nKey}.debug.buildStarted`, { projectName }));
+    } catch (err) {
+      logApiErrorInstance(err, new ApiErrorContext({ accountId, projectName }));
+      return;
+    }
+
+    await pollBuildStatus(accountId, projectName, currentBuild.get());
+    currentBuild.clear();
+
+    if (standbyeQueue.length > 0) {
+      await processStandByQueue(accountId, projectName);
+    }
+
+    queue.start();
+    logger.log(i18n(`${i18nKey}.logs.resuming`));
+  }, 5000);
+};
+
+const queueFileUpload = async (
+  accountId,
+  projectName,
+  filePath,
+  remotePath
+) => {
+  if (!isAllowedExtension(filePath)) {
+    logger.debug(i18n(`${i18nKey}.debug.extensionNotAllowed`, { filePath }));
+    return;
+  }
+  if (shouldIgnoreFile(filePath)) {
+    logger.debug(i18n(`${i18nKey}.debug.ignored`, { filePath }));
+    return;
+  }
+  if (!queue.isPaused) {
+    debounceQueueBuild(accountId, projectName);
+  }
+
+  logger.debug(i18n(`${i18nKey}.debug.uploading`, { filePath, remotePath }));
+
+  return queue.add(async () => {
+    await currentBuild.update(accountId, projectName);
+    try {
+      await uploadFileToBuild(
+        accountId,
+        projectName,
+        currentBuild.get(),
+        filePath,
+        remotePath
+      );
+      logger.log(
+        i18n(`${i18nKey}.logs.uploadSucceeded`, { filePath, remotePath })
+      );
+    } catch (err) {
+      logger.debug(
+        i18n(`${i18nKey}.debug.uploadFailed`, { filePath, remotePath })
+      );
+    }
+  });
+};
+
+const createNewBuild = async (accountId, projectName) => {
+  try {
+    logger.debug('Attempting to create a new build');
+    const { buildId } = await provisionBuild(accountId, projectName);
+    return buildId;
+  } catch (err) {
+    if (err.error.subCategory === 'PipelineErrors.PROJECT_LOCKED') {
+      logger.error('Project is locked, cannot create new build');
+    } else if (err.error.subCategory === 'PipelineErrors.MISSING_PROJECT') {
+      logger.error(`Project ${projectName} does not exist.`);
+    } else {
+      logApiErrorInstance(err, new ApiErrorContext({ accountId, projectName }));
+    }
+    process.exit(1);
+  }
+};
+
+exports.command = 'watch [path]';
+exports.describe = false;
+
+exports.handler = async options => {
+  loadAndValidateOptions(options);
+
+  const { path: projectPath } = options;
+  const accountId = getAccountId(options);
+
+  trackCommandUsage('project-watch', { projectPath }, accountId);
+
+  const { projectConfig, projectDir } = await getProjectConfig(projectPath);
+
+  validateProjectConfig(projectConfig, projectDir);
+
+  const watcher = chokidar.watch(path.join(projectDir, projectConfig.srcDir), {
+    ignoreInitial: true,
+    ignored: file => shouldIgnoreFile(file),
+  });
+
+  watcher.on('ready', () => {
+    logger.log(i18n(`${i18nKey}.logs.watching`, { projectPath }));
+  });
+  watcher.on('add', async filePath => {
```

Comment:
The first param here can be an array of strings. `add` and `change` look identical so I _think_ this can just be `watcher.on(['add', 'change']...`

Reply:
Is this documented somewhere? I don't see it in [the docs](https://github.com/paulmillr/chokidar) and in testing, that didn't work.

Reply:
Huh, yeah looks like you're right. At this point I forget where I initially saw it, but maybe I was mistakingly looking at the docs for one of the other `watcher` funcs.